File: tags/rel0.4.0/python/test/test_align.py
#    MolTK is a Python and C++ toolkit for protein sequence/structure alignment and visualization
#    Copyright (C) 2011  Christopher M. Bruns
#
#    This program is free software; you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation; either version 2 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License along
#    with this program; if not, write to the Free Software Foundation, Inc.,
#    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#   
#    Commercial users should ask about our dual licensing model.
#    For questions contact: cmbruns@rotatingpenguin.com

# Synopsis sketch of python API for moltk alignment, like sequoia
from moltk import Aligner, Sequence

aligner = Aligner()
alignment = aligner.align(Sequence("AACDEF"))
alignment = aligner.align(Sequence("ACEFG"))
print alignment

# Mirroring tutorial example 1
python                               # (no analogy)
from moltk import Aligner, Sequence  # (no analogy)
seq1 = Sequence(file("test1.seq"))   # read SEQ1 test1.seq
seq2 = Sequence(file("test2.seq"))   # read SEQ2 test2.seq
align = Aligner().align(seq1, seq2)  # ALIGN
print align                          # print ALIGN
seq1 = align                         # set SEQ1 ALIGN
seq2 = Sequence(file("test3.seq"))   # read SEQ2 test3.seq
align = Aligner().align(seq1, seq2)  # ALIGN
print align                          # print ALIGN
print align.id()                     # print id ALIGN
quit()                               # quit

# More compactly
from moltk import Aligner, Sequence
aligner = Aligner()
aligner.align(Sequence(file("test1.seq"))
aligner.align(Sequence(file("test2.seq"))
align = aligner.align(Sequence(file("test3.seq"))
print align
print align.id()

# Example 2
import moltk
from moltk import Aligner, Sequence, Structure
struct1 = Structure().load("test1.pdb")
struct2 = Structure().load("test2.pdb")
alignment = Aligner().align(struct1, struct2, moltk.ALIGN_SEQUENCE_ONLY)
overlay = Aligner.overlay(alignment)
# TODO dcutoff

File: tags/rel0.4.0/c++/test/test_align.cpp
/*
    MolTK is a Python and C++ toolkit for protein sequence/structure alignment and visualization
    Copyright (C) 2011  Christopher M. Bruns

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
    
    Commercial users should ask about our dual licensing model.
    For questions contact: cmbruns@rotatingpenguin.com
*/

// #define BOOST_TEST_DYN_LINK
#define BOOST_TEST_MODULE AlignerUnitTest
#include <boost/test/unit_test.hpp>
#include <boost/lexical_cast.hpp>
#include <boost/test/debug.hpp>
#include "moltk.hpp"
#include <iostream>
#include <cassert>


using namespace std;
using namespace moltk;
using moltk::units::bit;
using namespace boost;

BOOST_AUTO_TEST_CASE( test_aligner )
{
    boost::debug::detect_memory_leaks(false);
    // boost::debug::break_memory_alloc(566);
    Aligner aligner;
    aligner.set_end_gaps_free(true);
    Alignment a1 = aligner.align("W", "W");
    cout << a1 << a1.score() << endl;
    BOOST_CHECK_EQUAL(a1.score(), 5.5 * bit);

    aligner.set_end_gaps_free(false);
    Alignment a = aligner.align("QS", "SD");
    cout << a << a.score() << endl;
    aligner.set_end_gaps_free(true);
    a = aligner.align("QS", "SD");
    cout << a << a.score() << endl;

    Biosequence seq1("AACDEF");
    Biosequence seq2("ACEFG");
    a = aligner.align(seq1, seq2);
    cout << a << a.score() << endl;
}

